ledger-core/v2/vanilla/iokit/zerobytes.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package iokit
+
+import "io"
+
+var zeroBytes [1024]byte
+
+func WriteZeros(x int, w io.Writer) (int, error) {
+	total := 0
+	for ; x > len(zeroBytes); x -= len(zeroBytes) {
+		n, err := w.Write(zeroBytes[:])
+		total += n
+		if err != nil {
+			return total, err
+		}
+	}
+
+	n, err := w.Write(zeroBytes[:x])
+	total += n
+	return total, err
+}
+
+var _ io.ReadWriter = NullReadWriter{}
+var null = NullReadWriter{}
+
+type NullReadWriter struct{}
+
+func (p NullReadWriter) Read(b []byte) (int, error) {
+	for i := range b {
```

Comment:
why won't you do the same `copy` trick here? Copying of bytes should be cheaper here?

Reply:
because repeated store operation of a zero from a CPU register is faster than a memory-to-memory copy.
While zeroBytes are used to avoid allocation and zeroing of a memory buffer